src/Nest/CommonAbstractions/Infer/Field/FieldExpressionVisitor.cs

```diff
@@ -14,7 +14,9 @@ namespace Nest
 {
 	internal class HasConstantExpressionVisitor : ExpressionVisitor
 	{
-		public bool Found { get; private set; }
+		public bool Found => this.FoundConstants.Any(b => b);
```

Comment:
We're checking for any `true` values within the collection, so `Count > 0` isn't enough.